### ISSUE TITLE
router.events is not an Observable per docs

### ISSUE THREAD

**Issue by dudewad** [angular/angular#12473]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Also see here, I find the accepted answer here to not address the actual issue:
https://github.com/angular/angular/issues/10295

Using a very standard approach of filtering Router events, I get an error:
`router.events.filter is not a function`

Relevant Code:

	constructor(private router:Router){
		this.routerEventSub = router.events
			.filter(evt => evt instanceof NavigationStart)
			.subscribe(evt => {
				//do stuff
			});
	}

**Expected behavior**
I would expect the above implementation to "do stuff". Instead, I get the error 
`router.events.filter is not a function`

**Minimal reproduction of the problem with instructions**
The above implementation is sufficient to reproduce for me. THe strange thing is that it doesn't seem to reproduce in all cases but I cannot find a reason for why (two projects configured nearly identically provide different results).

**What is the motivation / use case for changing the behavior?**
Mainly, the docs say that Router.events is an Observable.
https://angular.io/docs/ts/latest/api/router/index/Router-class.html#!#events-anchor
However, logging `this.router.events` shows that it's a Subject.

**Please tell us about your environment:**
Windows 10, Intellij Idea Ultimate, NPM, using NG2 w/ Webpack/Webpack-dev-server.

* **Angular version:** 
2.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Chrome, FF, IE Edge
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TS 2.0.3 -> ES5

**Comment by DzmitryShylovich** [comment]
`import 'rxjs/add/operator/filter';`

**Comment by pkozlowski-opensource** [comment]
Thnx @DzmitryShylovich !

**Comment by dudewad** [comment]
@pkozlowski-opensource 
@DzmitryShylovich 
I'm sorry, clearly I'm not getting it. I referenced a ticket that had that exact same answer.
Does importing the filter operator magically turn the router.events Subject into an Observable? Because when I import that I get new errors.
What I'm asking for is clarification, and also an explanation as to why the docs say one thing and the reality is different.

**Comment by pkozlowski-opensource** [comment]
>  I referenced a ticket that had that exact same answer. Why you would give me the same answer again is confusing.

Please provide a minimal plunker with a reproduce scenario, otherwise it is confusing what your exact issue is....

**Comment by dudewad** [comment]
Ok, so I figured it out on my own. For anyone else who was confused on it, importing the filter operator anywhere will patch it globally (it seems).
A Subject _is both_ an observer and an observable. The issue is that if you aren't importing the operator it won't work, since importing the filter operator patches Observables (and therefore Subjects) to support the operator.

@pkozlowski-opensource This doesn't change the fact that the docs state that `router.events` is an Observable, not a Subject. This is confusing, particularly when the docs are so sparse, since developers effectively need to understand the rxjs Subject/Observable relationship.

Sorry, I'm not used to importing functionality on a class that I've already imported, that's why this is confusing.
